src/tv2_afvd_showstyle/helpers/pieces/graphicBackgroundLoop.ts

```diff
@@ -1,75 +1,73 @@
+import { GraphicsContent, PieceLifespan, TSR, WithTimeline } from 'blueprints-integration'
 import {
-	GraphicsContent,
-	IBlueprintActionManifest,
-	IBlueprintAdLibPiece,
-	IBlueprintPiece,
-	PieceLifespan,
-	TSR,
-	WithTimeline
-} from 'blueprints-integration'
-import { calculateTime, CueDefinitionBackgroundLoop, literal, ShowStyleContext, TV2ShowStyleConfig } from 'tv2-common'
+	calculateTime,
+	CueDefinitionBackgroundLoop,
+	DveLoopGenerator,
+	EvaluateCueResult,
+	literal,
+	ShowStyleContext,
+	TV2ShowStyleConfig
+} from 'tv2-common'
 import { SharedGraphicLLayer, SharedOutputLayer } from 'tv2-constants'
-import { CasparLLayer } from '../../../tv2_afvd_studio/layers'
 import { SourceLayer } from '../../layers'
 
 export function EvaluateCueBackgroundLoop(
 	context: ShowStyleContext,
-	pieces: IBlueprintPiece[],
-	adlibPieces: IBlueprintAdLibPiece[],
-	_actions: IBlueprintActionManifest[],
 	partId: string,
 	parsedCue: CueDefinitionBackgroundLoop,
 	adlib?: boolean,
 	rank?: number
-) {
+): EvaluateCueResult {
+	const result = new EvaluateCueResult()
 	const start = (parsedCue.start ? calculateTime(parsedCue.start) : 0) ?? 0
 
 	if (parsedCue.target === 'DVE') {
+		const dveLoopGenerator = new DveLoopGenerator() // todo: where to instantiate it?
```

Comment:
Unfortunately, we don't have dependency injection, nor is this method inside a class. So for now, I think this is an okay place for it (not great, but okay).

If you want you could also utilize a Facade that would then handle the creation of the DveLoopGenerator and then this method you are in would not have to care about the instantiation. That responsibility would be handled by the Facade.

But I'm okay with instantiating here for now.

Reply:
You could also do something similar to how `GfxSchemaGenerator` is instantiated in `getRundown.ts`